e2e/testcases/apiservice_test.go

```diff
@@ -17,36 +17,117 @@ package e2e
+	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/rbac.yaml", "acme/cluster/rbac.yaml")
 	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/apiservice.yaml", "acme/cluster/apiservice.yaml")
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush("add testing resources")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("adding apiservice resources")
 	nt.T.Log("Waiting for nomos to sync new APIService")
 	nt.WaitForRepoSyncs()
 
-	nt.T.Log("Removing APIService")
+	err := validateObjects(nt)
+	if err != nil {
+		nt.T.Fatalf("validate failed %s", err)
+	}
+
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/cluster/apiservice.yaml")
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove API service")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter API service")
+	nt.WaitForRepoSyncs()
+
+	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace-custom-metrics.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove custom metric stackdriver adapter namespace")
 	nt.WaitForRepoSyncs()
 }
 
-func TestImporterAndSyncerResilientToBadAPIService(t *testing.T) {
-	nt := nomostest.New(t)
-	nt.T.Log("Adding bad APIService")
+func TestImporterAndSyncerResilientToFlakyAPIService(t *testing.T) {
+	nt := nomostest.New(t, ntopts.RequireGKE(t), ntopts.Unstructured)
+	nt.T.Log("Adding API service with backend temporarily unavailable")
 	nt.MustKubectl("apply", "-f", "../testdata/apiservice/apiservice.yaml")
 	nt.T.Cleanup(func() {
 		nt.MustKubectl("delete", "-f", "../testdata/apiservice/apiservice.yaml", "--ignore-not-found")
+		t.Cleanup(func() {
+			if t.Failed() {
+				nt.PodLogs(adapterNamespace, adapterName, "pod-custom-metrics-stackdriver-adapter", true)
+			}
+		})
 	})
 
-	nt.T.Log("Creating commit with Deployment")
-	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/namespace-resilient.yaml", "acme/namespaces/resilient/namespace-resilient.yaml")
+	nt.T.Log("Creating commit with resources")
+	nt.RootRepos[configsync.RootSyncName].Copy("../testdata/apiservice/rbac.yaml", "acme/cluster/rbac.yaml")
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("add testing resources")
+
+	nt.T.Log("Adding backend")
+	nt.MustKubectl("apply", "-f", "../testdata/apiservice/namespace.yaml")
+	nt.MustKubectl("apply", "-f", "../testdata/apiservice/namespace-custom-metrics.yaml")
 	nt.T.Log("Waiting for nomos to stabilize")
 	nt.WaitForRepoSyncs()
+	err := validateObjects(nt)
+	if err != nil {
+		nt.T.Fatalf("validate failed %s", err)
+	}
+}
+
+func validateObjects(nt *nomostest.NT) error {
```

Comment:
given this package is large, maybe make this name more specific.

Reply:
Renamed.